src/PositionVelocityTime.jl

```diff
@@ -21,30 +23,127 @@ module PositionVelocityTime
+        lat = user_lla.lat
+        lon = user_lla.lon
+
+        ê = [-sind(lat), cosd(lat), 0]
+        n̂ = [-cosd(lon)*sind(lat), -sind(lat)*sind(lon), cosd(lat)]
+        û = [cosd(lon)*cosd(lat), sind(lon)*cosd(lat), sind(lat)]
+
+        el = asind(dot(p̂,û))
+        az = atand(dot(p̂,ê),dot(p̂,n̂))
+
+        return el, az
+
+    end
 
     """
     Calculates ECEF position of user
 
     $SIGNATURES
     ´sat_state´: satellite state, combining decoded data, code- and carrierphase 
+    
+    ´min_elevation´: the minimum elevation for a satellite, that can be used for a positioning
+                    min_elevation = -100, deactivates the  elevation filter
+    
 
     This function calculates the position of the user in ECEF coordinates
+        incl. earth rotation correction
+
+  
     The implementation follows IS-GPS-200K Table 20-IV.
     """
     function calc_PVT(
-        satellite_states::AbstractVector{SatelliteState{Float64}}
+        prev_pvt::PVTSolution,
+        satellite_states::AbstractVector{SatelliteState{Float64}},
+        min_elevation::Int64 = -100 #elevation filter is normally turned off
         )
-        
-        
-        usable_satellite_states = filter(x -> is_sat_healthy_and_decodable(x.decoder_state), satellite_states)
 
+        c = satellite_states[1].decoder_state.constants.c
+        usable_satellite_states = filter(x -> is_sat_healthy_and_decodable(x.decoder_state), satellite_states)
         length(usable_satellite_states) >= 4 || throw(SmallData("Not enough usable SV Data"))
         sv_positions = map(x -> sat_position_ECEF(x), usable_satellite_states)
-        pseudoranges = pseudo_ranges(usable_satellite_states)
+        
+        if prev_pvt.num_used_sats == 0
+            pseudoranges = pseudo_ranges(usable_satellite_states)
+            pvt_sol = user_position([0,0,0], sv_positions, pseudoranges)
+            prev_pos = pvt_sol.pos
+        else
+            prev_pos = prev_pvt.pos
+        end
+
+        # Remove Satellite with low elevation angle
+        if min_elevation >= 0 && min_elevation < 90 # min elevaiton filter
+            out = map( x-> get_elevation_azimuth(prev_pos, x), sv_positions)
+            el, az = map(x->getindex.(out,x), 1:2)
+
+            buf_state = usable_satellite_states
+            buf_sat_pos = sv_positions
+            el_save = Float64[]
+            az_save = Float64[]
+            PRN_save = Int64[]
+            for i = 1:length(usable_satellite_states)
+                if el[i] < min_elevation
+                    usable_satellite_states = filter(x -> buf_state[i].decoder_state.PRN != x.decoder_state.PRN, usable_satellite_states)
+                    sv_positions = filter(x -> x != buf_sat_pos[i], sv_positions)
+                else
+                    append!(PRN_save, buf_state[i].decoder_state.PRN)
+                    append!(el_save, el[i])
+                    append!(az_save, az[i])
+                end
+            end
```

Comment:
This can probably be simplified by using broadcasting
```suggestion
            el_filter = el .>= min_elevation
            usable_satellite_states = usable_satellite_states[el_filter]
            el_save = el[el_filter]
            az_save = az[az_filter]
            PRN_save = map(x.decoder_state.PRN, usable_satellite_states)
```

Reply:
I'm surprised that this works:
```julia
PRN_save = map(x.decoder_state.PRN, usable_satellite_states)
```
or is it a mistake and it should be?
```julia
PRN_save = map(x -> x.decoder_state.PRN, usable_satellite_states)
```

Reply:
You're right, changed it above